lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -32,384 +57,165 @@ def get(section: str, option: str, fallback=None) -> str:
-                strict: Boolean value to make sure path exist if True.
-
-            Returns:
-                Raising FileNotFoundError Exception
-            """
-            assert strict
-            raise FileNotFoundError(f"No such file or directory: '{self}'")
-
-        monkeypatch.setattr(Path, "glob", lambda path, args: self.tarlist)
-        monkeypatch.setattr(Path, "resolve", mock_resolve)
-        monkeypatch.setattr(Path, "stat", self.mock_stat)
-        obj = UnpackTarballs(MockConfig, make_logger)
-        result = obj.unpack_tarballs(min_size, max_size)
-        if result.total > 0:
-            for tar in self.tarlist:
-                assert (
-                    f"Tarball link, '{tar}', does not resolve to a real location: No such file or directory: '{tar}'"
-                    in caplog.text
-                )
-        assert result.total == total
-        assert result.success == 0
-
-    def test_unpack_md5_error(self, monkeypatch, make_logger, caplog):
-        """Show that when md5 file path cannot be accessed it throws a
-        FileNotFoundError Exception"""
-
-        mocks_called = []
-
-        def mock_md5sum(filepath: Path):
-            """
-            Mock of md5sum function
-
-            Args:
-                filepath: Tarball file path
-
-            Returns:
-                Raises FileNotFoundError Exception
-            """
-            mocks_called.append("mock_tarball_md5")
-            raise FileNotFoundError(f"No such file or directory: '{filepath}.md5'")
-
-        monkeypatch.setattr("pbench.server.utils.md5sum", mock_md5sum)
-        obj = UnpackTarballs(MockConfig, make_logger)
-        with pytest.raises(FileNotFoundError):
-            obj.unpack(self.tar, None)
-            assert (
-                f"Getting md5 value of tarball '{self.tar}' failed: No such file or directory: '{self.tar}.md5'"
-                in caplog.text
+        """Test that the tarbal list is filtered by the unpack bucket size
+        configuration."""
+
+        with monkeypatch.context() as m:
+            m.setattr("pbench.server.unpack_tarballs.Path", MockPath)
+            m.setattr("pbench.server.unpack_tarballs.Sync", MockSync)
+            m.setattr("pbench.server.unpack_tarballs.CacheManager", MockCacheManager)
+            m.setattr(Metadata, "getvalue", getvalue)
+
+            obj = UnpackTarballs(MockConfig(), make_logger)
+            result = obj.unpack_tarballs(min_size, max_size)
+            assert result.total == len(targets)
+            assert result.success == len(targets)
+            assert sorted(MockCacheManager.unpacked) == sorted(targets)
+            assert sorted(MockSync.record.keys()) == sorted(targets)
+            for id, actions in MockSync.record.items():
+                assert actions["did"] == Operation.UNPACK
+                assert actions["enabled"] == [Operation.COPY_SOS, Operation.INDEX]
+
+    @pytest.mark.parametrize("fail", [["md5.1"], ["md5.1", "md5.2"], []])
```

Comment:
I think we tested the "success case" at line 155, so it seems like the third argument here should be the "total fail" case (e.g., `["md5.1", "md5.2", "md5.3"]`), no?